evap/staff/tests/test_json_importer.py

```diff
@@ -0,0 +1,202 @@
+import json
+from datetime import date, datetime
+
+from django.test import TestCase
+from model_bakery import baker
+
+from evap.evaluation.models import Contribution, Course, Evaluation, Semester, UserProfile
+from evap.staff.importers.json import ImportDict, JSONImporter
+
+EXAMPLE_DATA: ImportDict = {
+    "students": [
+        {"gguid": "0x1", "email": "1@example.com", "name": "1", "christianname": "1"},
+        {"gguid": "0x2", "email": "2@example.com", "name": "2", "christianname": "2"},
+    ],
+    "lecturers": [
+        {"gguid": "0x3", "email": "3@example.com", "name": "3", "christianname": "3", "titlefront": "Prof. Dr."},
+        {"gguid": "0x4", "email": "4@example.com", "name": "4", "christianname": "4", "titlefront": "Dr."},
+    ],
+    "events": [
+        {
+            "gguid": "0x5",
+            "lvnr": 1,
+            "title": "Prozessorientierte Informationssysteme",
+            "title_en": "Process-oriented information systems",
+            "type": "Vorlesung",
+            "isexam": False,
+            "courses": [
+                {"cprid": "BA-Inf", "scale": "GRADE_PARTICIPATION"},
+                {"cprid": "MA-Inf", "scale": "GRADE_PARTICIPATION"},
+            ],
+            "relatedevents": {"gguid": "0x6"},
+            "appointments": [{"begin": "15.04.2024 10:15", "end": "15.07.2024 11:45"}],
+            "lecturers": [{"gguid": "0x3"}],
+            "students": [{"gguid": "0x1"}, {"gguid": "0x2"}],
+        },
+        {
+            "gguid": "0x6",
+            "lvnr": 2,
+            "title": "Prozessorientierte Informationssysteme",
+            "title_en": "Process-oriented information systems",
+            "type": "Klausur",
+            "isexam": True,
+            "courses": [
+                {"cprid": "BA-Inf", "scale": ""},
+                {"cprid": "MA-Inf", "scale": ""},
+            ],
+            "relatedevents": {"gguid": "0x5"},
+            "appointments": [{"begin": "29.07.2024 10:15", "end": "29.07.2024 11:45"}],
+            "lecturers": [{"gguid": "0x3"}, {"gguid": "0x4"}],
+            "students": [{"gguid": "0x1"}, {"gguid": "0x2"}],
+        },
+    ],
+}
+EXAMPLE_JSON = json.dumps(EXAMPLE_DATA)
+
+
+class TestImportUserProfiles(TestCase):
+    def setUp(self):
+        self.students = EXAMPLE_DATA["students"]
+        self.lecturers = EXAMPLE_DATA["lecturers"]
+
+        self.semester = baker.make(Semester)
+
+    def test_import_students(self):
+        self.assertEqual(UserProfile.objects.all().count(), 0)
+
+        importer = JSONImporter(self.semester)
+        importer._import_students(self.students)
+
+        user_profiles = UserProfile.objects.all()
+        self.assertEqual(user_profiles.count(), 2)
+
+        for i, user_profile in enumerate(user_profiles):
```

Comment:
* The loop already checks that there are no more user profiles (because it triggers an out-of-bound access on self.students otherwise), so we don't necessarily need the count assertion. The pattern is repeated below in other tests, I think I'd favor shorter/less test code if it covers the same aspects.

* I don't think you can assume that the user profiles will always be iterated in the same order. so I'd say the iterate-and-compare approach here is flaky. We could explicitly sort by `id` (which would rely on the creation order in the importer), or, possibly nicer, we could try to get some kind of set-comparison (i.e., assert that the set of user profiles is equal to a hardcoded expected set)